source/AI.cpp

```diff
@@ -1712,7 +1718,8 @@ void AI::MoveIndependent(Ship &ship, Command &command) const
 		vector<const StellarObject *> planets;
 		for(const StellarObject &object : origin->Objects())
 			if(object.HasSprite() && object.HasValidPlanet() && object.GetPlanet()->HasSpaceport()
-					&& object.GetPlanet()->CanLand(ship))
+					&& object.GetPlanet()->CanLand(ship) &&
+					(unrestricted || !gov->IsRestrictedFrom(*object.GetPlanet())))
```

Comment:
We should roll this `(unrestricted || !gov->IsRestrictedFrom(*object.GetPlanet())` part into Politics::CanLand (which is called by Planet::CanLand).